File: python/FruitBaskets.py
from math import factorial
from sys import stdin
from itertools import combinations


def at_least_200(x):
    return x >= 200


def choose(n, k):
    return factorial(n) // (factorial(k) * factorial(n-k))  # integer division avoids overflow


# Take in inputs
num = int(stdin.readline())
weights = [int(i) for i in stdin.readline().split(" ")]

# Check each combination of 1, 2, and 3 kinds of fruits
# Add the total weights that are at least 200
count = sum(filter(at_least_200, weights))

for i in range(2, num+1):
    if i < 4:
        count += sum(filter(at_least_200, [sum(lst) for lst in combinations(weights, i)]))
    else:
        # Since each kind of fruit weighs at least 50 grams,
        # for combinations of 4 or more fruits, we just count them all
        for weight in weights:
            # number of times each fruit appeared in nCr is (n-1)C(r-1)
            count += weight * choose(num-1, i-1)

print(count)

File: java/FruitBaskets.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

public class FruitBaskets {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int ct = 0;
        int[] weights = new int[n];
        for (int i = 0; i < n; i++) {
            weights[i] = sc.nextInt();
            if (weights[i] >= 200) ct += weights[i];
        }
        System.out.println(Arrays.toString(weights));

        // Since each kind of fruit weighs at least 50 grams,
        // for combinations of 4 or more fruits, we just count them all

    }

    private static ArrayList<int[]> combinations(int start, int end, int n, int k) {
        ArrayList<int[]> arr = new ArrayList<>();
        if (k == 1) {
            for (int i = start; i < end; i++) {
                arr.add(new int[]{i});
            }
            return arr;
        }
        for (int[] comb: combinations(start+1, end, n-1, k-1)) {

        }
        return arr;
    }

}
